src/irr/asset/CImageLoaderBMP.cpp

```diff
@@ -202,42 +202,43 @@ asset::IAsset* CImageLoaderBMP::loadAsset(io::IReadFile* _file, const asset::IAs
 
 	_file->read(&header, sizeof(header));
 	
-	byte BitmapHeader[header.BitmapHeaderSize];
+	std::vector<uint8_t> BitmapHeader;
+	_file->read(&BitmapHeader,header.BitmapHeaderSize);
 	
-		_file->read(&BitmapHeader, sizeof(header.BitmapHeaderSize));
-	byte Extra[16];
+	uint8_t Extra[16];
 	int i = 0;
 	//extra is 4 dwords 
 	//icc is the icc size 
-		
-	#ifdef _DEBUG
-		os::Printer::log("Extra bits", ELL_DEBUG);
-	#endif
 
-	while(i<16){
-		byte Extra[i] =  BitmaphHeader[(header.BitmapHeaderSize+i)];
+#ifdef _DEBUG
+	os::Printer::log("Extra bits", ELL_DEBUG);
+#endif
+
+	while (i < 16) {
+		 Extra[i] = BitmapHeader[(header.BitmapHeaderSize + i)];
 		i++;
-			#ifdef _DEBUG
-				os::Printer::log(std::to_wstring( Extra[i]), ELL_DEBUG);
-			#endif
+#ifdef _DEBUG
+		os::Printer::log(std::to_wstring(Extra[i]), ELL_DEBUG);
+#endif
 	};
- 	byte ICC_size =  BitmaphHeader[(header.BitmapHeaderSize-2)];	
-	byte ICC[ICC_size];
-	i = 0;
+	uint8_t ICC_size = BitmapHeader.at((header.BitmapHeaderSize - 2));
```

Comment:
Use `operator[]` instead of `at()` - faster. at() checks if you're not going out of range and throws if so